### ISSUE TITLE
z_mean and z_log_var in variational_autoencoder.py

### ISSUE THREAD

**Issue by clockwiser** [fchollet/keras#7404]
z_mean and z_log_var look like same.
.....
x = Input(batch_shape=(batch_size, original_dim))
h = Dense(intermediate_dim, activation='relu')(x)
z_mean = Dense(latent_dim)(h)
z_log_var = Dense(latent_dim)(h)
.....
How is it possible?

**Comment by augustkx** [comment]
Hi, I have the same question as you. Do you get the answer?

**Comment by clockwiser** [comment]
They are different, because each get through different layer of same level of the neural network. 
By optimization, z_mean will be close to mean, and z_log_var be close to std.

**Comment by augustkx** [comment]
Hi, thanks very much for the reply and the link. 
But can I ask you another question: for the Custom loss layer, there're slight difference between your link and https://github.com/fchollet/keras/blob/master/examples/variational_autoencoder_deconv.py
for this: xent_loss = original_dim * metrics.binary_crossentropy(x, x_decoded_mean), why original_dim is multiplied here?

And, for the link you provided, why there are just sums , no means, shouldn't
 'recon = K.sum(K.binary_crossentropy(y_pred, y_true), axis=1)'
be sum and then mean?

I will  appreciate it very much if you could reply.

**Comment by clockwiser** [comment]
See the source of metrics.binary_crossentropy. It returns mean of the cross entropy.

def binary_crossentropy(y_true, y_pred):
    return K.mean(K.binary_crossentropy(y_true, y_pred), axis=-1)
